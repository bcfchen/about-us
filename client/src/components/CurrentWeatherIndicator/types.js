import PropTypes from "prop-types";
import HourlyForecast from "../../models/HourlyForecast";

export const propTypes = {
    isLoading: PropTypes.bool.isRequired,
    currentHourlyForecast: PropTypes.instanceOf(HourlyForecast)
};
