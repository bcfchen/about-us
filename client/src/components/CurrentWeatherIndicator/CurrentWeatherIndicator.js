import React from "react";
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const CurrentWeatherIndicator = ({ currentCondition, location }) => {
    return (
        <div className="current-weather-indicator-container">
            <Typography variant="display2" gutterBottom>
                {location.getLocationString()}
            </Typography>
            <Typography variant="display1" gutterBottom>
                {currentCondition.text}
            </Typography>
            <Typography variant="display1" gutterBottom>
                {currentCondition.temp}
            </Typography>
        </div>
    );
};

export default CurrentWeatherIndicator;