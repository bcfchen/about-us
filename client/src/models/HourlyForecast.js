import moment from "moment";

class HourlyForecast {
    constructor(properties, location) {
        this.temperature = properties.Temperature.Value;
        this.text = properties.IconPhrase;
        this.dateTime = properties.DateTime;
        this.windSpeed = properties.Wind.Speed.Value + " km/h";
        this.humidity = properties.RelativeHumidity + " %";
        this.uvIndex = properties.UVIndex;
        this.location = location;
    }

    getHoursAndMinutes() {
        return moment(this.dateTime).format("HH:mm A");
    }

    getLocationString() {
        return this.location.getLocationString();
    }
}

export default HourlyForecast;