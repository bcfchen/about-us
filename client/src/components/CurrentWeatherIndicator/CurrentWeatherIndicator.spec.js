import React from "react";
import { shallow } from "enzyme";
import CurrentWeatherIndicator from "./CurrentWeatherIndicator";
import HourlyForecast from "../../models/HourlyForecast";

describe("<CurrentWeatherIndicator/>", () => {
    let defaultWeatherProps = {
        Temperature: {
            Value: 15
        },
        Wind: {
            Speed: {
                Value: 6
            }
        },
        IconPhrase: "Sunny"
    };
    let defaultProps = {
        isLoading: true,
        currentHourlyForecast: new HourlyForecast(defaultWeatherProps)
    };

    let component = null;
    let render = (props = defaultProps) => {
        component = shallow(<CurrentWeatherIndicator {...props} />)
    }

    describe("on page load", () => {
        it("should render correctly with default properties", () => {
            render();
            expect(component).toMatchSnapshot();
        });

        describe("contains current forecast property", () => {
            beforeEach(() => {
                render();
            });
            it("should display the temperature correctly", () => {
                expect(component.find(".current-temperature").dive().dive().text()).toEqual("15");
            });

            it("should display the weather description correctly", () => {
                expect(component.find(".weather-description").dive().dive().text()).toEqual("Sunny");
            });
        });

        describe("does not contain current forecast property", () => {
            beforeEach(() => {
                render({ isLoading: true, currentHourlyForecast: null });
            });
            it("should not display the temperature", () => {
                expect(component.find(".current-temperature").dive().dive().text()).toEqual("");
            });

            it("should not display the weather description", () => {
                expect(component.find(".weather-description").dive().dive().text()).toEqual("");
            });
        });
    });
});