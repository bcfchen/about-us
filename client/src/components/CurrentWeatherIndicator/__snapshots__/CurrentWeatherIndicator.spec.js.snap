// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentWeatherIndicator/> on page load should render correctly with default properties 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrentWeatherIndicator
    currentHourlyForecast={
        HourlyForecast {
            "dateTime": undefined,
            "humidity": "undefined %",
            "isDaylight": undefined,
            "location": undefined,
            "temperature": 15,
            "text": "Sunny",
            "uvIndex": undefined,
            "windSpeed": "6 km/h",
          }
    }
    isLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="current-weather-indicator-container"
>
        <WithStyles(Typography)
                className="current-temperature"
                variant="display4"
        >
                15
        </WithStyles(Typography)>
        <WithStyles(Typography)
                className="weather-description"
                gutterBottom={true}
                variant="headline"
        >
                Sunny
        </WithStyles(Typography)>
</div>,
      "in": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            className="current-temperature"
            variant="display4"
>
            15
</WithStyles(Typography)>,
          <WithStyles(Typography)
            className="weather-description"
            gutterBottom={true}
            variant="headline"
>
            Sunny
</WithStyles(Typography)>,
        ],
        "className": "current-weather-indicator-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 15,
            "className": "current-temperature",
            "variant": "display4",
          },
          "ref": null,
          "rendered": 15,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Sunny",
            "className": "weather-description",
            "gutterBottom": true,
            "variant": "headline",
          },
          "ref": null,
          "rendered": "Sunny",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="current-weather-indicator-container"
>
          <WithStyles(Typography)
                    className="current-temperature"
                    variant="display4"
          >
                    15
          </WithStyles(Typography)>
          <WithStyles(Typography)
                    className="weather-description"
                    gutterBottom={true}
                    variant="headline"
          >
                    Sunny
          </WithStyles(Typography)>
</div>,
        "in": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="current-temperature"
              variant="display4"
>
              15
</WithStyles(Typography)>,
            <WithStyles(Typography)
              className="weather-description"
              gutterBottom={true}
              variant="headline"
>
              Sunny
</WithStyles(Typography)>,
          ],
          "className": "current-weather-indicator-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 15,
              "className": "current-temperature",
              "variant": "display4",
            },
            "ref": null,
            "rendered": 15,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Sunny",
              "className": "weather-description",
              "gutterBottom": true,
              "variant": "headline",
            },
            "ref": null,
            "rendered": "Sunny",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
